nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1330 @@
+
+        Returns
+        -------
+        BrainModel
+        """
+        if isinstance(other, BrainModel):
+            if self.affine is None:
+                affine, shape = other.affine, other.volume_shape
+            else:
+                affine, shape = self.affine, self.volume_shape
+                if other.affine is not None and (
+                        (other.affine != affine).all() or
+                        other.volume_shape != shape
+                ):
+                    raise ValueError("Trying to concatenate two BrainModels defined " +
+                                     "in a different brain volume")
+            nvertices = dict(self.nvertices)
+            for name, value in other.nvertices.items():
+                if name in nvertices.keys() and nvertices[name] != value:
+                    raise ValueError("Trying to concatenate two BrainModels with inconsistent " +
+                                     "number of vertices for %s" % name)
+                nvertices[name] = value
+            return type(self)(
+                    np.append(self.name, other.name),
+                    np.concatenate((self.voxel, other.voxel), 0),
+                    np.append(self.vertex, other.vertex),
+                    affine, shape, nvertices
+            )
+        return NotImplemented
+
+    def __getitem__(self, item):
+        """
+        Extracts part of the brain structure
+
+        Parameters
+        ----------
+        item : anything that can index a 1D array
+
+        Returns
+        -------
+        If `item` is an integer returns a tuple with 3 elements:
+        - boolean, which is True if it is a surface element
+        - vertex index if it is a surface element, otherwise array with 3 voxel indices
+        - structure.BrainStructure object describing the brain structure the element was taken from
+
+        Otherwise returns a new BrainModel
+        """
+        if isinstance(item, integer_types):
+            return self.get_element(item)
+        if isinstance(item, string_types):
+            raise IndexError("Can not index an Axis with a string (except for Parcels)")
+        return type(self)(self.name[item], self.voxel[item], self.vertex[item],
+                          self.affine, self.volume_shape, self.nvertices)
+
+    def get_element(self, index):
+        """
+        Describes a single element from the axis
+
+        Parameters
+        ----------
+        index : int
+            Indexes the row/column of interest
+
+        Returns
+        -------
+        tuple with 3 elements
+        - boolean, which is True if it is a surface element
+        - vertex index if it is a surface element, otherwise array with 3 voxel indices
+        - structure.BrainStructure object describing the brain structure the element was taken from
+        """
+        is_surface = self.name[index] in self.nvertices.keys()
+        name = 'vertex' if is_surface else 'voxel'
```

Comment:
Rather than using `getattr` below, I'd suggest using:

```suggestion
        struct = self.vertex if is_surface else self.voxel
```